frontend/src/views/Collection/components/DatasetTab/index.tsx

```diff
@@ -50,12 +49,9 @@ const DatasetTab: FC<Props> = ({
   const isDatasetPresent =
     datasets?.length > 0 || Object.keys(uploadedFiles).length > 0;
 
-  const invalidateCollectionQuery = memoize(
-    () => {
-      queryClient.invalidateQueries([USE_COLLECTION, collectionId, visibility]);
-    },
-    () => collectionId + visibility
-  );
+  const invalidateCollectionQuery = () => {
+    queryClient.invalidateQueries([USE_COLLECTION, collectionId]);
```

Comment:
removing visibility here fixes [2381](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/2381)

Reply:
Ah maybe we can keep memoize and just remove `visibility`?

```ts
const invalidateCollectionQuery = memoize(
    () => {
      queryClient.invalidateQueries([USE_COLLECTION, collectionId]);
    },
    () => collectionId
  );
```

This way we don't create a new function every time we render, and forces `CollectionDatasetsGrid` and `DatasetsGrid` to re-render as well ♼!